pynest/examples/pynest-microcircuit/network.py

```diff
@@ -0,0 +1,347 @@
+                self.dc.append(dc)
+            print 'DC generator created'
+
+    # connections between neuronal populations are created in this function
+    def create_connections(self):
+        mean_delays = self.net_dict['mean_delay_matrix']
+        std_delays = self.net_dict['std_delay_matrix']
+        norm_clp = 'normal_clipped'
+        for i, target_pop in enumerate(self.pops):
+            for j, source_pop in enumerate(self.pops):
+                synapse_nr = int(self.synapses_scaled[i][j])
+                if synapse_nr >= 0.:
+                    weight = self.weight_mat[i][j]
+                    w_sd = abs(weight * self.weight_mat_std[i][j])
+                    conn_dict_poisson = {'rule': 'all_to_all'}
+                    syn_dict_poisson = {'model': 'static_synapse',
+                                        'weight': self.w_ext,
+                                        'delay': {'distribution': norm_clp,
+                                                  'mu': 1.5,
+                                                  'sigma': 0.75,
+                                                  'low': self.sim_resolution}}
+                    conn_dict_rec = {'rule': 'fixed_total_number',
+                                     'N': synapse_nr}
+                    syn_dict_rec_ex = {'model': 'static_synapse',
+                                       'weight': {'distribution': norm_clp,
+                                                  'mu': weight,
+                                                  'sigma': w_sd,
+                                                  'low': 0.0},
+                                       'delay': {'distribution': norm_clp,
+                                                 'mu': mean_delays[i][j],
+                                                 'sigma': std_delays[i][j],
+                                                 'low': self.sim_resolution}}
+                    syn_dict_rec_in = {'model': 'static_synapse',
+                                       'weight': {'distribution': norm_clp,
+                                                  'mu': weight,
+                                                  'sigma': w_sd,
+                                                  'high': 0.0},
+                                       'delay': {'distribution': norm_clp,
+                                                 'mu': mean_delays[i][j],
+                                                 'sigma': std_delays[i][j],
+                                                 'low': self.sim_resolution}}
+                    if j % 2:
+                        syn_dict_rec = syn_dict_rec_in
+                    else:
+                        syn_dict_rec = syn_dict_rec_ex
+                    nest.Connect(source_pop, target_pop,
+                                 conn_spec=conn_dict_rec,
+                                 syn_spec=syn_dict_rec)
+        print 'Recurrent connections established'
```

Comment:
I prefer to put such print statements in the beginning of the functions for keeping track of where time is spent, e.g.,: `Establishing recurrent connections`